### DISCUSSION TITLE
Comments box during the payment screen.

### DISCUSSION THREAD

**Discussion by ghost** [btcpayserver/btcpayserver#4563] [Ideas / Feature Requests]
I want to suggest a "comments box" during the payment screen. 

When a user is on the payment screen. I would like to leave directions to him. 

In my case I want him to write his username inside the comments box so when the payment confirms i can just look on the invoice and see the comments and know who paid the invoice.  

Currently I leave directions on the previous screen to email me with their username so I know which user paid. 
Sometimes the user forgets to email me and wonders why i haven't upgraded his account but the issue is I dont know who bought the subscription. 

![Screenshot_2023-01-24_11-40-57](https://user-images.githubusercontent.com/31393558/214384882-405fef65-2308-4e34-b1bf-00f8317f3abf.jpg)

**Comment by dennisreimann** [comment]
Does the new [form feature introduced in v1.7](https://blog.btcpayserver.org/btcpay-server-1-7-0/#-form-builder-request-shipping-address) help?

**Reply by ghost** [reply]
Where is the form feature located? I am looking for it in the settings but can't seem to locate it.

**Reply by ghost** [reply]
Oh i found it. It's 

Enable Checkout v2 by going to Store Settings > Checkout appearance and flip the switch!

I'm going to enable it and try it out

**Reply by ghost** [reply]
So i enabled it and I noticed there is 2 different options to enable. 

A form that requires them to enter their shipping info such as name, address etc.
A form that requires them to enter their email only. 

I enabled the second form because I sell digital items and only require their username. 
It would be great if we could edit those forms so I could change the text "email"  to "username"

It works for now. 

![EmailBtcPayServer](https://user-images.githubusercontent.com/31393558/215620343-95a8561f-1925-4443-aa1b-b5ec7f5a1684.jpg)

**Reply by dstrukt** [reply]
Understandable, and more custom input is on the roadmap - we appreciate your feedback!

**Comment by pavlenex** [comment]
We've added more custom inputs in 1.8.0 and UI for forms is coming in 1.9.10 most likely.
